### PR TITLE
Handle labels in outer scope in control flow pass

### PR DESCRIPTION
**Pull request by jjonescz**
Fixes https://github.com/dotnet/roslyn/issues/73068.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs [hunk 1]
```diff
@@ -550,6 +550,62 @@ .maxstack  2
...
+        {
+            var code = """
+                x:
+                System.Action a = () =>
+                {
+                    using System.IDisposable d = null;
+                    goto x;
+                };
+                """;
+            CreateCompilation(code).VerifyDiagnostics(
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs:564]: >VerifyDiagnostics [](http://example.com/codeflow?start=36&length=17)

VerifyEmitDiagnostics?

File: src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs [hunk 2]
```diff
@@ -550,6 +550,62 @@ .maxstack  2
...
+        {
+            var code = """
+                System.Action a = () =>
+                {
+                    using System.IDisposable d = null;
+                    goto x;
+                };
+                x:;
+                """;
+            CreateCompilation(code).VerifyDiagnostics(
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs:584]: >VerifyDiagnostics [](http://example.com/codeflow?start=36&length=17)

VerifyEmitDiagnostics?

File: src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs [hunk 3]
```diff
@@ -550,6 +550,62 @@ .maxstack  2
...
+        public void GotoInLambda_NonExistent()
+        {
+            var code = """
+                System.Action a = () =>
+                {
+                    using System.IDisposable d = null;
+                    goto x;
+                };
+                """;
+            CreateCompilation(code).VerifyDiagnostics(
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/CodeGen/GotoTest.cs:603]: >VerifyDiagnostics [](http://example.com/codeflow?start=36&length=17)

VerifyEmitDiagnostics?

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs [hunk 4]
```diff
@@ -7889,6 +7889,118 @@ Element Values(0)
...
+                                            Predecessors (0)
+                                            Statements (0)
+                                    }
+                        Next (Regular) Block[B2]
+                            Leaving: {R1}
+                }
+                Block[B2] - Exit
+                    Predecessors: [B1]
+                    Statements (0)
+                """;
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs:7991]: nit: consider left-aligning so it's easier to see those long lines

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1) with minor comment

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Pull request merged**